pkg/controller/instance/instance_controller.go

```diff
@@ -466,38 +334,15 @@ func parameterDifference(old, new map[string]string) map[string]string {
 	return diff
 }
 
-// newPlanExecution creates a PlanExecution based on the kind and instance for a given planName
-func newPlanExecution(instance *kudov1alpha1.Instance, planName string, scheme *runtime.Scheme) (*kudov1alpha1.PlanExecution, error) {
-	gvk, err := apiutil.GVKForObject(instance, scheme)
-	if err != nil {
-		return nil, err
-	}
+type executionError struct {
```

Comment:
`executionError` is confusing to me... many things could be an execution error... I'm not sure when to use it and how it is different from InstanceError.   The places it is used it looks like a PlanExecutionError.   perhaps it is fine... but I had to hunt for it's use to understand it which means like some challenge in readability.

and... although we collapse planexec into instance... I could still see having a planexecution package which would allow for planexec.NewError etc. or plan... like plan.NewError